lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py

```diff
@@ -362,24 +408,35 @@ def create_update_containerinstance(self):
             image = container_def.get("image")
             memory = container_def.get("memory", 1.5)
             cpu = container_def.get("cpu", 1)
+            commands = container_def.get("commands")
             ports = []
+            variables = []
 
             port_list = container_def.get("ports")
             if port_list:
                 for port in port_list:
                     ports.append(self.cgmodels.ContainerPort(port=port))
 
+            variable_list = container_def.get("environment_variables")
+            if variable_list:
```

Comment:
what if update an existing environemtn variable or remove one?

Reply:
it's full overwrite

Reply:
current container instance implementation has no proper idempotency as well.
this is one of the things that need to be handled properly and it's a large change which I mentioned today.